### PR TITLE
479 Risk management arrangements

### PR DESCRIPTION
**Pull request by Harriethw**
# Context

https://trello.com/c/IkDDCcmZ/479-rosh-risk-management-arrangements

# Changes in this PR

The 'Risk management arrangements' question within RoSH.
This has a new pattern for us - checkboxes with a 'none of the above' type option. I've added custom to javascript so that if 'no' is checked, the other options are unchecked and vice versa. It will also wipe any text entered in the 'details' text input, as we do not want to risk accidentally surfacing data that should not be saved or shared.

## Screenshots of UI changes

Screenshot 2023-09-18 at 14 00 23

Screenshot 2023-09-18 at 14 00 33

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Review by patrickjfl — APPROVED** [review]
Nice 🙌 

Just wondering whether we need the extra javascript, or could make do with informing the user via error messaging what combinations of answers they can submit?

File: server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk [hunk 1]
```diff
@@ -86,4 +86,40 @@
...
+      document
+        .getElementById('arrangements')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+      document
+        .getElementById('arrangements-2')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+      document
+        .getElementById('arrangements-3')
+        .addEventListener('click', () => setCheckedToFalseByIds(['arrangements-5']));
+    });
```
Comment (edavey) [inline server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk:123]: Is this a scenario where selecting by class could add clarity and reliability?

Am I right to think we could end up with something simpler like e.g. 

```
    document.addEventListener('DOMContentLoaded', function () {
      // 'no' checkbox sets the others to unchecked
      document
        .getElementsByClassName('no-arrangements')
        .addEventListener('click', () => {
          clearAnySelectedArrangements()
        });
      // clicking any arrangement checkbox sets 'no' to unchecked
      document
        .getElementsByClassName('arrangement')
        .addEventListener('click', () => resetNoArrangementsChoice());
    });
```

Apologies in advance if I'm way off the mark here!

**Review by edavey — APPROVED** [review]
Nice work! 🏅 Just a couple of tiny ideas/suggestions

File: server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk (hunk 1 above)
Comment (Harriethw) [reply server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk:123]: thanks Ed, that's a really good shout!

Comment (Harriethw) [reply server/views/applications/pages/risk-of-serious-harm/risk-management-arrangements.njk]: unfortunately it didn't seem possible to add a `class` to an `item` according to the nunjucks options https://design-system.service.gov.uk/components/checkboxes/, but I've set a data attribute that will achieve a similar thing and allow us to select all three in one go.

File: integration_tests/tests/apply/risks_and_needs/risk-of-serious-harm/risk_management_arrangements.cy.ts [hunk 2]
```diff
@@ -15,6 +15,14 @@
...
 //    And I continue to the next task / page
 //    Then I see the "cell share information" page
+//
+//  Scenario: Toggle between selected arrangements and no arrangements
+//    Given I have selected that there are arrangements
+//    When I select that this person does not have any arrangements
+//    Then none of the other checkboxes are marked as selected
+//    And the arrangement details are cleared
+//    When I select an arrangement
+//    Then the 'no arrangements' option is unselected
```
[the remark this answers is not in the document]
Comment (Harriethw) [reply integration_tests/tests/apply/risks_and_needs/risk-of-serious-harm/risk_management_arrangements.cy.ts]: good point! I struggle to get my head round writing the behaviour wording but it makes sense to me when I see it :)

**Pull request merged**